### PR TITLE
MINOR: Don't ignore deletion of partition metadata file and log topic id clean-ups

### PR DESCRIPTION
**Pull request by ijuma**
Log if deletion fails and don't expose log topic id for mutability outside of `assignTopicId()`.

Also remove an unnecessary parameter in `PartitionTest`.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
cc @jolshan

**Review by jolshan — APPROVED** [review]
LGTM. I like that we are preventing direct access to writing the topic ID field.

File: core/src/main/scala/kafka/log/Log.scala [hunk 1]
```diff
@@ -325,19 +325,25 @@ class Log(@volatile private var _dir: File,
     // Ensure we do not try to assign a provided topicId that is inconsistent with the ID on file.
     if (partitionMetadataFile.exists()) {
         if (!keepPartitionMetadataFile)
-          partitionMetadataFile.delete()
+          try partitionMetadataFile.delete()
+          catch {
+            case e: IOException =>
+              error(s"Error while trying to delete partition metadata file ${partitionMetadataFile}", e)
```
Comment (chia7712) [inline core/src/main/scala/kafka/log/Log.scala]: Should `partitionMetadataFile` override `toString` to offer readable message?

**Review by chia7712 — APPROVED** [review]
+1 and one minor comment below.

File: core/src/main/scala/kafka/log/Log.scala (hunk 1 above)
Comment (jolshan) [reply core/src/main/scala/kafka/log/Log.scala]: Ah this is a good point. The toString method could include the file path.

Comment (ijuma) [reply core/src/main/scala/kafka/log/Log.scala]: Good point, addressed.

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by ijuma** [comment]
JDK 11 build passed, a few unrelated failures for the other builds:

> Build / JDK 8 and Scala 2.12 / testMetricsDuringTopicCreateDelete() – kafka.integration.MetricsDuringTopicCreationDeletionTest
> 16s
> Build / JDK 15 and Scala 2.13 / testMetricsDuringTopicCreateDelete() – kafka.integration.MetricsDuringTopicCreationDeletionTest
> 7s
> Build / JDK 15 and Scala 2.13 / testCreateClusterAndCreateAndManyTopicsWithManyPartitions() – kafka.server.RaftClusterTest
> 19s
> Build / JDK 15 and Scala 2.13 / testCreateClusterAndWaitForBrokerInRunningState() – kafka.server.RaftClusterTest
> 1m 11s
> Build / JDK 15 and Scala 2.13 / testCreateClusterAndCreateListDeleteTopic() – kafka.server.RaftClusterTest
> 16s
> Build / JDK 15 and Scala 2.13 / testCreateClusterAndCreateListDeleteTopic() – kafka.server.RaftClusterTest

**Pull request merged**